docs/recipes/Authentication.md

```diff
@@ -0,0 +1,1447 @@
+
+<details>
+  <summary>Example session storage implementation</summary>
+
+```typescript title="/app/utils/storage/SessionStorage.ts"
+import { MMKV } from "react-native-mmkv"
+
+const storage = new MMKV({
+  id: "session",
+})
+
+// TODO: Remove this workaround for encryption: https://github.com/mrousavy/react-native-mmkv/issues/665
+storage.set("workaround", true)
+
+/**
+ * A simple wrapper around MMKV that provides a base API
+ * that matches AsyncStorage for use with Supabase.
+ */
+
+/**
+ * Get an item from storage by key
+ *
+ * @param {string} key of the item to fetch
+ * @returns {Promise<string | null>} value for the key as a string or null if not found
+ */
+export async function getItem(key: string): Promise<string | null> {
+  try {
+    return storage.getString(key) ?? null
+  } catch {
+    console.warn(`Failed to get key "${key}" from secure storage`)
+    return null
+  }
+}
+
+/**
+ * Sets an item in storage by key
+ *
+ * @param {string} key of the item to store
+ * @param {string} value of the item to store
+ */
+export async function setItem(key: string, value: string): Promise<void> {
+  try {
+    storage.set(key, value)
+  } catch {
+    console.warn(`Failed to set key "${key}" in secure storage`)
+  }
+}
+
+/**
+ * Removes a single item from storage by key
+ *
+ * @param {string} key of the item to remove
+ */
+export async function removeItem(key: string): Promise<void> {
+  try {
+    storage.delete(key)
+  } catch {
+    console.warn(`Failed to remove key "${key}" from secure storage`)
+  }
+}
+```
+
+</details>
+
+### Encrypting the User Session
+
+If you'd like to encrypt the user's session because it [contains sensitive information](https://reactnative.dev/docs/security#storing-sensitive-info), you can take advantage of Expo SecureStore and MMKV's encryption. Expo SecureStore will securely store key-value pairs locally on device in the iOS keychain or Android Keystore. The reason we need both is that [Expo SecureStore has a size limit of 2048 bytes](https://docs.expo.dev/versions/latest/sdk/securestore/). The Supabase session is already larger than 2048 bytes by default so we're going to generate a unique key with Expo Crypto to encrypt the Session Store with MMKV and store that key with Expo SecureStore.
+
+First, we'll need to install those additional dependencies:
+
+```bash title="Terminal"
+bunx expo install expo-secure-store expo-crypto
+```
+
+We'll also need to add Expo SecureStorage to our [plugin configuration](https://docs.expo.dev/versions/latest/sdk/securestore/#configuration-in-appjsonappconfigjs):
+
+```typescript title="/app.config.ts"
+  return {
+    ...config,
+    plugins: [
+      ...existingPlugins,
+      // success-line
+      "expo-secure-store",
```

Comment:
I would add this to the static config, `app.json` instead (it'll come in through `...existingPlugins` and you'll find the more standard ones from third party packages that aren't coded inside Ignite there.

splash screen and flipper disabled ones are here because you can't require in app.json

Reply:
🤦‍♂️ My initial implementation had it there and then my recipe evaluation didn't. Good find! 🙌